pkg/agent/downloader.go

```diff
@@ -0,0 +1,85 @@
+package agent
+
+import (
+	"fmt"
+	pro "github.com/kubeflow/kfserving/pkg/agent/protocols"
+	"hash/fnv"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type Downloader struct {
+	// TODO: Add back-off retries
+	ModelDir   string
+	ProtocolManagers map[pro.Protocol]pro.ProtocolManager
+}
+
+var SupportedProtocols = []pro.Protocol{pro.S3}
+
+func (d *Downloader) DownloadModel(event EventWrapper) error {
+	modelSpec := event.ModelSpec
```

Comment:
If nothing else is using event's other fields, why not just pass ModelSpec directly into this method?

Reply:
I will now need the modelName